pkg/dataobj/sections/logs/reader.go

```diff
@@ -179,9 +179,15 @@ func (r *Reader) Read(ctx context.Context, batchSize int) (arrow.Record, error)
 			}
 
 			columnBuilder := builder.Field(columnIndex)
+			columnType := r.opts.Columns[columnIndex].Type
 
 			if val.IsNil() {
-				columnBuilder.AppendNull()
+				if columnType == ColumnTypeMessage {
+					// Message columns are required, so return an empty value instead of a NULL value.
+					columnBuilder.(*array.StringBuilder).BinaryBuilder.AppendEmptyValue()
```

Comment:
This is fine, but you might want to consider marking the column as non-nullable in the field schema (look at `columnToField`) and then check for that here instead:

```go
if !builder.Schema().Field(columnIndex).Nullable {
  columnBuilder.AppendEmptyValue()
} else {
  columnBuilder.AppendNull()
}
```

(ApendEmptyValue is defined on the general builder interface so you don't need to cast it to the proper type)